src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionStatus.tsx

```diff
@@ -31,9 +31,9 @@ const TransactionStatus: FC<TransactionStatusProps> = ({
     >
       {pending && (
         <SpinnerGap
-          className="ml-2.5 h-[0.8125rem] w-[0.8125rem] animate-spin text-blue-400"
-          size={14}
-        />
+        className="absolute right-[-20px] top-[0.125rem] h-[0.8125rem] w-[0.8125rem] animate-spin text-blue-400"
+        size={14}
+      />
```

Comment:
Thanks for opening a PR to address issue #2520  @codeseker 😄 

I think the component being referred to in the issue is the `SpinnerGap` component found in `src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/GroupedTransactionStatus.tsx`

Looking forward to test your fix again!

Reply:
Hey @codeseker, the component that needs to be fixed is in `src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/GroupedTransactionStatus.tsx` and it's called `SpinnerGap`.

I think you're making changes to the wrong file based on your latest changes:
![Screenshot 2024-06-13 at 14 23 45](https://github.com/JoinColony/colonyCDapp/assets/50642296/7935e70e-820c-4263-8da6-85de3470b08d)

Reply:
Hi @codeseker , it seems that the spinner now disappears due to the style updates:
![image](https://github.com/JoinColony/colonyCDapp/assets/50642296/8429897b-1054-4a66-8cbf-433d61e3eec9)

The requirement is to have the spinner rendered within the status pill itself ref [#2520](https://github.com/JoinColony/colonyCDapp/issues/2520):
![image](https://github.com/JoinColony/colonyCDapp/assets/50642296/0ce41879-755d-4abc-978c-f7a226d745be)

I think `position: absolute` is not the best solution to this since it wouldn't respect the styling constraints of the pill. A better approach would be to either:

Pass in the `SpinnerGap` component to the `PillsBase` component as a loading component then toggle it via another prop to indicate that the status is pending (or have it so that the `PillsBase` component automatically infers that the status is pending if a loading component is passed)
```js
<PillsBase
  className={clsx({
    'bg-success-100 text-success-400': succeeded,
    'bg-negative-100 text-negative-400': failed,
    'bg-gray-100 text-gray-500': !succeeded && !failed,
  })}
  loadingComponent={<SpinnerGap
    className="ml-2.5 h-[0.8125rem] w-[0.8125rem] animate-spin text-blue-400"
    size={14}
  />}
  isPending={isPending}
>
  {status.toLowerCase()}
</PillsBase>;

```
OR (my preferred approach) add the `SpinnerGap` component inside the `PillsBase` component by default and toggle it via the `isPending` prop
```js
<PillsBase
  className={clsx({
    'bg-success-100 text-success-400': succeeded,
    'bg-negative-100 text-negative-400': failed,
    'bg-gray-100 text-gray-500': !succeeded && !failed,
  })}
  isPending={isPending}
>
  {status.toLowerCase()}
</PillsBase>;
```

Reply:
Can you also please post a screenshot of the UI after your changes have been applied? Thank you :)

Reply:
The issue is resolved now